components/org.wso2.identity.apps.taglibs.layout.controller/src/main/java/org/wso2/identity/apps/taglibs/layout/controller/compiler/resolvers/FileResolver.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.apps.taglibs.layout.controller.compiler.resolvers;
+
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.CompilerException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * File resolver implementation for both local and remote files.
+ */
+public class FileResolver implements Resolver {
+
+    private InputStream inputStream = null;
+    private InputStreamReader streamReader = null;
+    private BufferedReader reader = null;
+
+    /**
+     * Get the reader to read the layout file.
+     *
+     * @param file Layout file path as a URL object.
+     * @return Reader to read the layout file.
+     */
+    @Override
+    public Reader getReader(URL file) {
+
+        if (file == null) {
+            throw new CompilerException("Can't find the given file");
+        }
+        try {
+            inputStream = file.openStream();
```

Comment:
Since you don't have a final block for the try-catch, the input stream will not be closed. Use try with resources instead of using the final block. Refer https://stackoverflow.com/a/28074565/10055162

Reply:
But according to this scenario we can't use a try with resource. Since I want to use this reader until the compiling process end. That is the reason I used a separate  close method to do that.